### DISCUSSION TITLE
where to setup allowed redirectUrl list in appwrite console

### DISCUSSION THREAD

**Discussion by bryanprimus** [appwrite/appwrite#7970] [Q&A]
Hi i'm currently working with magic url login and it requires us to setup a redirect url to redirect back to our app with token in queryParams.

when i'm providing my url, it tells me that it should only be one of this listed options
`localhost`
`cloud.appwrite.io`
`appwrite.io`

![image](https://github.com/appwrite/appwrite/assets/46083126/f64d6bf9-3c1f-4c84-96bd-addadc7c545f)

i assume that this is the domain that are allowed by default.
The problem is i'm unable to locate where to update this default configuration and add my domain or redirect to the list.

i have checked every section in console, but still couldn't locate it
![image](https://github.com/appwrite/appwrite/assets/46083126/bcb8b1bb-c78f-4361-9ae4-b42522ec2796)

i'm expecting a form like this 

![image](https://github.com/appwrite/appwrite/assets/46083126/de9c8e24-0f4f-45b9-b452-58987ca7fa72)

but i  couldn't find it anywhere in the console.
the screenshot is from supabase dashboard btw

**Comment by stnguyen90** [comment]
You need to add a Web Platform. See https://appwrite.io/docs/quick-starts/web#step-1

**Comment by HaneenAyman23** [comment]
What you’re seeing is expected behavior in Appwrite — those domains (localhost, cloud.appwrite.io, appwrite.io) are not a “redirect whitelist UI”, but a platform origin validation constraint enforced by design.
Key point
For Magic URL (and other auth flows), Appwrite does not use a separate redirect URL allowlist like Supabase.
Instead, redirect eligibility is derived from registered Platforms.

Why you're not seeing your domain
If your domain is not appearing/accepted, it means it has not been added as a Platform in the project.
Go to:
Console → Project → Platforms → Add Platform
Then:
Add Web app
Set your production domain (e.g. yourdomain.com)
For local dev, use localhost
Once added, Appwrite will allow redirects originating from that platform context.

Important clarification
The list you’re seeing:
locahost
cloud.appwrite.io
appwrite.io
represents default trusted origins / examples in the UI context, not a fixed global redirect whitelist you can edit.
There is no separate “redirect URL configuration screen” like Supabase.
How Magic URL redirect actually works
Magic URL uses:
Platform origin (registered in console)
SDK-provided redirect URL in the request
Appwrite session validation
So the redirect is validated implicitly, not via a manual whitelist.

**Comment by deepshekhardas** [comment]
To add to what's been shared: Appwrite doesn't have a standalone "allowed redirect URLs" list like some other auth providers. The redirect validation is tied to your **registered Platforms** in the console.

**Step-by-step fix:**
1. Go to **Appwrite Console → Your Project → Overview → Platforms**
2. Click **"Add Platform"**
3. Select **Web** (for web apps) or the appropriate platform
4. Add your domain (e.g., `yourdomain.com` for production, `localhost` for local dev)

Once your domain is registered as a Platform, Appwrite will accept redirect URLs originating from that domain in your Magic URL flow.

**Why this happens:** Appwrite validates the `redirect` param against the registered platform origins, not a separate URL allowlist. So the error `URL host must be one of: localhost, cloud.appwrite.io, appwrite.io` is telling you your app's domain isn't registered as a Platform yet.

**For Magic URL specifically**, make sure the `url` you pass to `account.createMagicURLToken()` has the same origin as your registered Platform:
```js
await account.createMagicURLToken(
  ID.unique(),
  'user@example.com',
  'https://yourdomain.com/auth/callback' // must match registered platform
);
```

Hope this clears it up!
